File: Erlang/asciiart.erl
% asciiart Erlang
-module(prog).
-export([main/0]).

main() ->
    for(-12, 12, fun(Y) ->
        for(-39, 39, fun(X) ->
            CA = X * 0.0458,
            CB = Y * 0.08333,
            mandel(0, CA, CB, CA, CB)
        end),
        io:fwrite("~n")
    end).

for(M, M, F) -> [F(M)];
for(I, M, F) -> [F(I) | for(I+1, M, F)].

mandel(16, _, _, _, _) ->
	io:fwrite(" ");
mandel(I, A, B, CA, CB) ->
	A2 = A * A - B * B + CA,
	B2 = 2 * A * B + CB,
	if
		(A2 * A2 + B2 * B2) > 4 ->
			io:fwrite(erlang:integer_to_list(I, 16));
		true ->
			mandel(I+1, A2, B2, CA, CB)
	end.

File: OCaml/asciiart.ml
(* asciiart OCaml (ocamlopt 4.05.0) *)
let mandel x y =
	let ca = (float_of_int x) *. 0.0458 in
	let cb = (float_of_int y) *. 0.08333 in
	let a = ref ca in
	let b = ref cb in
	let i = ref 0 in
	let cont = ref true in
	while !cont do
		let t = !a *. !a -. !b *. !b +. ca in
		b := 2.0 *. !a *. !b +. cb;
		a := t;
		if (!a *. !a +. !b *. !b) > 4.0 then begin
			Printf.printf "%X" !i;
			cont := false
		end else begin
			i := !i + 1;
			if !i > 15 then begin
                print_char ' ';
				cont := false
			end
		end
	done;
;;
let main () =
	for y = -12 to 12 do
		for x = -39 to 39 do
			mandel x y
		done;
    	print_newline ()
	done
;;
main ()
